engine/src/execution_graph/logic_controllers/taskflow/graph.cpp

```diff
@@ -250,6 +251,28 @@ namespace cache {
 			return std::set<graph::Edge>();
 		}
 	}
+
+	// This function will work when the Relational Algebra only contains: LogicalTableScan and LogicalLimit
+	void graph::check_how_data_looks() {
+		auto first_iterator = container_.begin(); // points to null
+		first_iterator++;
+		int32_t min_index_valid = first_iterator->first;
+		size_t total_kernels = container_.size(); 
+
+		if (total_kernels == 4) { // null, TableScanKernel, LimitKernel, OutputKernel
+			if ( get_node(min_index_valid )->expression == "OutputKernel" &&
+				 get_node(min_index_valid + 1)->get_type_id() == kernel_type::LimitKernel &&
+			 	 get_node(min_index_valid + 2)->get_type_id() == kernel_type::TableScanKernel )
```

Comment:
this should be TableScan OR BindableTableScan

Reply:
Done!